src/extension/features/budget/display-total-monthly-goals/index.js

```diff
@@ -0,0 +1,109 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { getEmberView } from 'toolkit/extension/utils/ember';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { getCurrentRouteName } from 'toolkit/extension/utils/ynab';
+
+export class DisplayTotalMonthlyGoals extends Feature {
+  shouldInvoke() {
+    return getCurrentRouteName().indexOf('budget') !== -1 && this.settings.enabled !== '0';
+  }
+
+  extractCategoryGoalInformation(element) {
+    const emberId = element.id;
+    const viewData = getEmberView(emberId).data;
+    const {
+      subCategory,
+      monthlySubCategoryBudgetCalculation
+    } = viewData;
+
+    const goalType = subCategory.get('goalType');
+    const monthlyFunding = subCategory.get('monthlyFunding');
+    const targetBalanceDate = monthlySubCategoryBudgetCalculation.get('goalTarget');
+
+    let monthlyGoalAmount = 0;
+
+    switch (goalType) {
+      case 'MF': {
+        monthlyGoalAmount = monthlyFunding;
+        break;
+      }
+      case 'TBD': {
+        monthlyGoalAmount = targetBalanceDate;
+        break;
+      }
+    }
+
+    return {
+      monthlyGoalAmount,
+      isChecked: $(element).is('.is-checked')
+    };
+  }
+
+  calculateMonthlyGoals() {
+    const categoryGoals = {
+      total: 0,
+      checkedTotal: 0,
+      checkedCount: 0
+    };
+
+    $('.budget-table-row.is-sub-category').each((index, element) => {
+      const categoryGoal = this.extractCategoryGoalInformation(element);
+
+      categoryGoals.total += categoryGoal.monthlyGoalAmount;
+      if (categoryGoal.isChecked) {
+        categoryGoals.checkedTotal += categoryGoal.monthlyGoalAmount;
+        categoryGoals.checkedCount++;
+      }
+    });
+
+    return {
+      amount: categoryGoals.checkedCount > 0
+        ? categoryGoals.checkedTotal
+        : categoryGoals.total,
+      checkedCategoryCount: categoryGoals.checkedCount
+    };
+  }
+
+  createInspectorElement(goalsAmount) {
+    const currencyClass = (goalsAmount === 0) ? 'zero' : 'positive';
+
+    return $(`
+      <div class="total-monthly-goals-inspector">
+        <h3>TOTAL MONTHLY GOALS</h3>
+        <h1 title>
+          <span class="user-data currency ${currencyClass}">
+            ${formatCurrency(goalsAmount)}
+          </span>
+        </h1>
+        <hr />
+      </div>
+    `);
+  }
+
+  invoke() {
+    const monthlyGoals = this.calculateMonthlyGoals();
+
+    $('.total-monthly-goals-inspector').remove();
+
+    const shouldShowInspector = monthlyGoals.checkedCategoryCount !== 1;
```

Comment:
is this for debugging? it seems like this feature isn't going to do anything if you've selected 0 or >1 category?

Reply:
@joshmadewell If you've selected zero categories, it shows the sum total goals for all categories.  If you've selected more than one, it shows the sum for those you selected.  This line prevents it from rendering the inspector if you've selected exactly one category, since the other budget inspector things don't show up in that scenario.

Let me know if there's a way you think this could be written more clearly.